third_party/proton/csrc/lib/Driver/GPU/Hip.cpp

```diff
@@ -0,0 +1,47 @@
+#include "Driver/GPU/Hip.h"
+#include "Driver/Dispatch.h"
+#include <hip/hip_runtime_api.h>
+#include <string>
+
+namespace proton {
+
+namespace hip {
+
+struct ExternLibHip : public ExternLibBase {
+  using RetType = hipError_t;
+  static constexpr const char *name = "libamdhip64.so";
+  static constexpr RetType success = hipSuccess;
+  static void *lib;
+};
+
+void *ExternLibHip::lib = nullptr;
+
+DEFINE_DISPATCH(ExternLibHip, deviceSynchronize, hipDeviceSynchronize)
+
+DEFINE_DISPATCH(ExternLibHip, deviceGetAttribute, hipDeviceGetAttribute, int *,
+                hipDeviceAttribute_t, int);
+
+Device getDevice(uint64_t index) {
+  int clockRate;
+  (void)hip::deviceGetAttribute<true>(&clockRate, hipDeviceAttributeClockRate,
+                                      index);
+  int memoryClockRate;
+  (void)hip::deviceGetAttribute<true>(&memoryClockRate,
+                                      hipDeviceAttributeMemoryClockRate, index);
+  int busWidth;
+  (void)hip::deviceGetAttribute<true>(&busWidth,
+                                      hipDeviceAttributeMemoryBusWidth, index);
+  int smCount;
+  (void)hip::deviceGetAttribute<true>(
+      &smCount, hipDeviceAttributeMultiprocessorCount, index);
+
+  // TODO: Compute capability is a NVIDIA concept. It doesn't map naturally to
+  // AMD GPUs. Figure out a better way to support this.
+  uint64_t arch = 0;
```

Comment:
Moving forward, we could change it to `std::string`, but it requires additional modifications in the viewer. I'm OK with arch=0 now

Reply:
Yeah makes sense. Don't feel this is really used for now. But it's good information to keep in the trace.